src/pages/wirkung.js

```diff
@@ -0,0 +1,37 @@
+import Head from 'next/head';
+import {useRouter} from 'next/router';
+import {useEffect, useState} from 'react';
+
+import Layout from '../components/Layout';
+import {db} from '../db';
+
+export default function WirkungPage() {
+	const router = useRouter();
+	const [ids, setId] = useState(0);
+	useEffect(() => {
+		setId(router.query.keyword);
+	}, [router.query]);
+
+	const entry = db.find(entry => entry.id === Number(ids));
```

Comment:
Ich habe gute Nachrichten für dich 🎉
An dieser Stelle benötigst du weder ein `useState()` noch ein `useEffect()`, denn alle "Informationen" die wir brauchen stecken `in const router = useRouter();`

Du kannst es wie folgt schreiben: 
```suggestion
	const router = useRouter();
    const {id} = router.query;

	const entry = db.find(entry => entry.id === Number(id));
```

Um diese Änderung besser nachvollziehen zu können, kannst du:

-  gerne noch einmal in das [Repo des Fish Shops, Branch: 02-nextjs-routing](https://github.com/aclaes/nf-nextjs-fish-shop/tree/02-nextjs-routing) schauen
- und die entsprechende Folge[ "Next.js Routing "von  ALEX-TV](https://neuefische-students.slack.com/archives/C03JYC8R89Z/p1659452859126079) ansehen 😁

Reply:
Ich verstehe nicht ganz, wenn alles, was wir benötigen, in useRouter steckt, weshalb lösche ich die Datei useRouter ? Bitte um Hilfe ... Ich verstehe noch nicht, welchen Weg wir gehen.

Reply:
- Habe meinen Ordner und meine Datei für dynamische Route angelegt. Ok, erledigt.